codegen/src/main/twirl/templates/JavaServer/Handler.scala.txt

```diff
@@ -24,18 +24,17 @@ import akka.grpc.javadsl.GrpcExceptionHandler;
 import static @{service.packageName}.@{service.name}.Serializers.*;
 
 public class @{service.name}HandlerFactory {
+  private static CompletionStage<HttpResponse> notFound = CompletableFuture.completedFuture(
+    HttpResponse.create().withStatus(StatusCodes.NOT_FOUND));
+
   public static Function<HttpRequest, CompletionStage<HttpResponse>> create(@service.name implementation, Materializer mat) {
     return (req -> {
       Iterator<String> segments = req.getUri().pathSegments().iterator();
       if (segments.hasNext() && segments.next().equals(@{service.name}.name) && segments.hasNext()) {
         String method = segments.next();
         return handle(req, method, implementation, mat).exceptionally(e -> GrpcExceptionHandler.standard(e));
-      }
-      else
-      {
-        CompletableFuture<HttpResponse> result = new CompletableFuture<>();
-        result.completeExceptionally(new UnsupportedOperationException("Unexpected path"));
-        return result;
+      } else {
+        return notFound;
```

Comment:
changed this to NOT_FOUND because I guess that's a more valid response if a single handler is used, and felt better to use that as the signal to continue with next handler in `concat` rather than using exception